core/chains/evm/client/client.go

```diff
@@ -98,8 +104,8 @@ var _ Client = (*client)(nil)
 
 // NewClientWithNodes instantiates a client from a list of nodes
 // Currently only supports one primary
-func NewClientWithNodes(logger logger.Logger, cfg PoolConfig, primaryNodes []Node, sendOnlyNodes []SendOnlyNode, chainID *big.Int) (*client, error) {
-	pool := NewPool(logger, cfg, primaryNodes, sendOnlyNodes, chainID)
+func NewClientWithNodes(logger logger.Logger, cfg PoolConfig, primaryNodes []Node, sendOnlyNodes []SendOnlyNode, chainID *big.Int, chainType config.ChainType) (*client, error) {
```

Comment:
Why pass chaintype separate instead of expanding PoolConfig?

Reply:
My thought was that chainID was also not included, so probably we didn't want to add chain-specific configs in the `PoolConfig` interface.

Reply:
Hmm but isn't the whole `PoolConfig` already chain specific?
What do the callers look like? e.g. it would seem silly to do `f(cfg, cfg.A(), cfg.B()` if we can just pass `f(cfg)`, no? If we don't get that simplification, then nbd :shrug:

Reply:
Actually, that's exactly how the callers are written 😛 :
https://github.com/smartcontractkit/chainlink/blob/bci-835_remove_err_dependencies_from_txm/core/chains/evm/chain.go#L94 .
Don't get me wrong, I'm up for extending PoolConfig I just want to understand why `chainID` was excluded in the first place.

Reply:
I wouldn't be surprised if it was separated just for testing convenience or something like that :shrug: . I'm not a fan of modifying production APIs just to accommodate writing test code, but it is a pattern that we have all over core - maybe a consequence of mocking interfaces in most places.

Reply:
Yeah, that makes sense. Let me try implementing it locally as a draft and see if I hit any roadblocks.

Reply:
Which PoolConfig are we talking about? This one?
`type PoolConfig interface {
	NodeSelectionMode() string
	NodeNoNewHeadsThreshold() time.Duration
}`

Doesn't seem chain specific.

Also, we do have a future goal to make the multi Node logic work for all chains, so the whole Pool and Nodes design needs to. be generalized for all chains. But don't expect this PR to start thinking about that.

So for the above, lets just do whatever is acceptable for this PR.

Reply:
It is chain specific: https://github.com/smartcontractkit/chainlink/blob/develop/docs/CONFIG.md#evmnodepool
This set of node pool configuration fields were explicitly set for a particular chain, or fallback to defaults. There is no global node pool config.